### ISSUE TITLE
Requestbody x-www-form-urlencoded is not properly rendered

### ISSUE THREAD

**Issue by lspvic** [swagger-api/swagger-ui#4643]
### Q&A (please complete the following information)
 - OS: windows
 - Browser: chrome
 - Version: 67
 - Method of installation: dist
 - Swagger-UI version: 3.17.0
 - Swagger/OpenAPI version: OpenAPI 3.0

 ### Content & configuration

Swagger/OpenAPI definition:
 ```yaml
      requestBody:
        description: user info to be filled
        content:
          application/x-www-form-urlencoded:
            schema:
              type: object
              required:
                - username
                - name
              properties:
                username:
                  type: string
                  description: user identity
                name:
                  type: string
                  description: user name
                phone_number:
                  type: string
                  pattern: '/\d{11}/'
                  description: phone number
 ```
![_20180613215633](https://user-images.githubusercontent.com/4335658/41356192-f486a800-6f55-11e8-9d7e-6bcaebdfe587.png)
 Swagger-UI configuration options:
 ```js
 SwaggerUI({
   // your config options here
 })
 ```

 ```
 ?yourQueryStringConfig
 ```

### Is your feature request related to a problem?
* property descriptions are not shown
* required properties are not starred
* required properties checking,  string regex pattern checking is not performed when try swagger client

However, parameters are ideally shown. There should be no much difference between parameters and body form data.

**Comment by arlix85** [comment]
Hello, I was about to open a new issue but I've seen this one which is correlated. I would like to integrate Ispvic report with the object-level "example" (as documented here https://swagger.io/docs/specification/adding-examples/#request-response-bodies) which doesn't seem to be supported:

```yaml
         application/x-www-form-urlencoded:
            schema:
              type: object
              properties:
                userid:
                  type: string
                  description: the id (or email address) inserted by the user in the login form
                password:
                  type: string
                  format: password
              example:
                userid: test
                password: password
```

Plus, the example is still populated by using the notation inside the properties...
```yaml
              properties:
                userid:
                  type: string
                  description: the id (or email address) inserted by the user in the login form
                  example: test
```
...but in the form the example is always rendered between quotation marks:
![screenshot-gimme5 acomea prv-2018 06 14-12-25-56](https://user-images.githubusercontent.com/36543409/41406924-22f6f0f0-6fce-11e8-9a0b-19db9ba6159c.png)

If something is wrong on my YAML code, please let me know, I wasn't able to correct the issue even with different notations.

Thanks,
Fabio

**Comment by lspvic** [comment]
@arlix85 , what is your required configuration for your schema, and which version is your swagger-ui. In my example, the red star and `required` is not shown.
Thank you.

**Comment by arlix85** [comment]
@lspvic just like "example", "required" is still working if used inside the property, like the old v2 standard:
```yaml
              properties:
                userid:
                  type: string
                  description: the id (or email address) inserted by the user in the login form
                  example: test
                  required: true
```

**Comment by lspvic** [comment]
@arlix85  In OpenAPI 3.0, required is not suitable for properties, there's a required array for properties, like
```yaml
         type: object
              required:
                  - userid
              properties:
                userid:
                  type: string
                  description: the id (or email address) inserted by the user in the login form
                  example: test
```
Thanks.

**Comment by arlix85** [comment]
@lspvic yes, I know, but since it's still working in SwaggerUI, I'm using this workaround until this issue is fixed.

**Comment by danielflower** [comment]
In case this helps, it's easy to reproduce.

Go to https://petstore.swagger.io/#/pet/updatePetWithForm
Then use this URL: https://gist.githubusercontent.com/danielflower/5c5ae8a46a0a49aee508690c19b33ada/raw/b06ff4d9764b5800424f6a21a40158c35277ee65/petstore.json

You can see endpoints such as `POST /pet/{petId}` rendering with `"string"`

These values are also the default in the Try It Out form, and annoyingly even if you clear the value it normally re-appears on blur.

**Comment by bestmike007** [comment]
@danielflower the default value issue is fixed since https://github.com/swagger-api/swagger-ui/commit/8f65483510770ef548ea7a79d9e5ba9a3387dcee#diff-47300e7ecd8989b6246221de88fc9a3cR633

The re-appear issue is definitely a bug.

**Comment by bestmike007** [comment]
I'm not sure which is the designed behavior:
*  The input value is set to `string` only once when clicked the "Try it out" button. In this case we should use a typeof check or `=== undefined` check here: https://github.com/swagger-api/swagger-ui/blob/v3.17.3/src/core/plugins/oas3/components/request-body.jsx#L98
*  The input value is set to `string` every time it's cleared. In such case we might use the placeholder property.

I prefer not setting the sample value at all if not present, use the description as the input placeholder would be better.

I've also noticed that all the parameter rows are re-render when one of them is changed, which might be a performance issue.

**Comment by sryze** [comment]
Is this planned to be fixed? I saw this issue a few months ago, thought that I would be addressed by now but after trying 3.0 again it seems to be present to this day.

**Comment by shockey** [comment]
This has been prioritized, and will be fixed shortly 😄

**Comment by shockey** [comment]
Fixed!

@lspvic, be advised that the forthcoming version of Swagger UI will break the property-level `required` that you mentioned.

**Comment by huihe** [comment]
Got the same issue here.

**Comment by shockey** [comment]
@huihe, this is fixed, are you sure you're on the latest version of Swagger UI? 😄

**Comment by yottachen** [comment]
I tried to use Swagger-UI 3.18.2(https://github.com/swagger-api/swagger-ui/releases). The description is still not shown when using urlencoded. 

And I notice that the `tag`(3.18.2) is inconsistent with the last commit(3.18.1). Is it proper tag?

**Comment by shockey** [comment]
@danielflower, are you still seeing this in latest (3.19.0)?

**Comment by danielflower** [comment]
That version doesn't appear to be available in maven central yet. (Using `org.webjars:swagger-ui`)

**Comment by deki** [comment]
It's available now: http://central.maven.org/maven2/org/webjars/swagger-ui/3.19.0/

**Comment by danielflower** [comment]
Works great now. Thanks a lot!

**Comment by shockey** [comment]
Great!

I'm going to close out this thread now - if anyone is having issues with this in the future, please open a new issue.
